plugins/catalog/src/apis/EntityPresentationApi/DefaultEntityPresentationApi.ts

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity, stringifyEntityRef } from '@backstage/catalog-model';
+import {
+  CatalogApi,
+  EntityPresentationApi,
+  EntityRefPresentation,
+  EntityRefPresentationSnapshot,
+} from '@backstage/plugin-catalog-react';
+import { HumanDuration } from '@backstage/types';
+import DataLoader from 'dataloader';
+import ExpiryMap from 'expiry-map';
+import uniq from 'lodash/uniq';
+import ObservableImpl from 'zen-observable';
+import {
+  DEFAULT_BATCH_DELAY,
+  DEFAULT_CACHE_TTL,
+  DEFAULT_ENTITY_FIELDS,
+  createDefaultRenderer,
+} from './defaults';
+import { durationToMs } from './util';
+
+/**
+ * A custom renderer for the {@link DefaultEntityPresentationApi}.
+ *
+ * @public
+ */
+export interface DefaultEntityPresentationApiRenderer {
+  /**
+   * An extra set of fields to request for entities from the catalog API.
+   *
+   * @remarks
+   *
+   * You may want to specify this to get additional entity fields. The smaller
+   * the set of fields, the more efficient requests will be to the catalog
+   * backend.
+   *
+   * The default set of fields is: apiVersion, kind, the scalar metadata fields
+   * (uid, etag, name, namespace, title, description), spec.type, and
+   * spec.profile.
+   *
+   * This field is ignored if async is set to false.
+   */
+  extraFields?: string[];
```

Comment:
Skipping fields for now, just fetching the entire entity to make sure that we can use entities reliably later on

Reply:
fixed in [54cbdc1](https://github.com/backstage/backstage/pull/17474/commits/430f2931b30efb373ea341b9c93f2f02054cbdc1)